File: Java/src/main/java/com/hamaraBasket/NonExpiryItemQualityChecker.java
package com.hamaraBasket;

public class NonExpiryItemQualityChecker extends QualityChecker {
    @Override
    public void updateQuality(HamaraBasketItem item) {
        if (item.getQuality() >= 50) {
            return;
        }
        int quality = item.getQuality();
        item.setQuality(quality + 1);
    }
}

File: CSharpCore/HamaraBasketQualityAssurance/NonExpiryItemQualityChecker.cs
namespace HamaraBasketQualityAssurance
{
    public class NonExpiryItemQualityChecker : QualityChecker
    {
        public override void UpdateQuality(HamaraBasketItem item)
        {
            if (item.Quality >= 50)
            {
                return;
            }
            item.Quality += 1;
        }
    }
}
